src/awesome/tag.rs

```diff
@@ -132,7 +135,30 @@ fn set_activated<'lua>(lua: &'lua Lua, (obj, val): (AnyUserData<'lua>, bool))
         }
         tag.activated = val;
     }
-    if !val {
+    let activated_tags = lua.named_registry_value::<Table>(TAG_LIST)?;
+    let activated_tags_count = activated_tags.len()?;
+    if val {
+        let index = activated_tags_count + 1;
+        activated_tags.set(index, obj.clone())?;
+    } else {
+        // Find and remove the tag in/from the list of tags
+        {
+            let tag_ref = &*obj.borrow::<TagState>()? as *const _;
+            let mut found = false;
+            for pair in activated_tags.clone().pairs::<Integer, AnyUserData>() {
```

Comment:
Is this clone necessary as well?

Reply:
This one is necessary. `pairs()` consumes the table reference, meaning that I would have to fetch a reference from the table from the registry again without it (so that I can remove the entry from the table). Cloning the reference seems like the better option.